### ISSUE TITLE
"Custom Block" that lists latest blog post of certain page by slug or id?

### ISSUE THREAD

**Issue by clientwestwood** [PiranhaCMS/piranha.core#493]
Using Piranha CMS 4.x (core 5.3.0) for a while. Pretty neat.

Trying to create a "Custom Block" that lists latest blog post of certain page by slug or id.

Have anybody succeed doing it?

I am not an advanced dotnet player so please detailed answers are much appreciated.

**Comment by clientwestwood** [comment]
For example I use this:
```
@model Piranha.Models.BlogArchive

<div class="main">
    @foreach (var post in Model.Archive.Posts){
        <h2><a href="@post.Permalink">@post.Title</a></h2>
    }
</div>
```
But getting this error:
`InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'Piranha.Extend.Blocks.ArchiveBlock', but this ViewDataDictionary instance requires a model item of type 'Piranha.Models.BlogArchive'.`

@tidyui any help appreciated :)

**Comment by rengert** [comment]
Page models and block models are different stories.

In this case the block is using the piranha.extend.block.archiveblock and passes it to the view. But in your view you defined the model to be another type (the type normally used if you write the view for the page, not for the block).

Please update the first line to

@model Piranha.Extend.Blocks.ArchiveBlock;

You should update the for loop accordingly.

**Comment by tidyui** [comment]
Do you have the latest versions of the packages?

    <PackageReference Include="Piranha" Version="5.3.1" />
    <PackageReference Include="Piranha.AspNetCore" Version="5.3.1" />
    <PackageReference Include="Piranha.AspNetCore.Identity.SQLite" Version="5.3.0" />
    <PackageReference Include="Piranha.AttributeBuilder" Version="5.3.0" />
    <PackageReference Include="Piranha.ImageSharp" Version="5.3.0-rc1" />
    <PackageReference Include="Piranha.Local.FileStorage" Version="5.3.0" />
    <PackageReference Include="Piranha.Manager" Version="5.3.1" />

**Comment by tidyui** [comment]
The reason I'm asking is that there's a bug in `5.3.0` where `WebApp.Api` is not set which was fixed in `5.3.0`. This would explain your error message. Make sure you have the latest packages **or** change the DisplayTemplate in the web project to the following:

    @model RazorWeb.Models.Blocks.ArchiveBlock
    @inject Piranha.IApi Api
    @{
        var archive = Model.GetArchive<RazorWeb.Models.BlogArchive>(Api);
    }

    @foreach (var post in archive.Archive.Posts)
    {
        <a href="@post.Permalink"><h1>@post.Title</h1></a>
    }

This will **inject** the `IApi` from container into the view.

Best regards

**Comment by clientwestwood** [comment]
It works! @tidyui This is literally best dotnet cms ever :)
